.github/workflows/main.yml

```diff
@@ -0,0 +1,223 @@
+name: Continuous Integration
+
+on: [push]
+
+jobs:
+  multiple-pythons:
+    name: ${{ matrix.os }} with Python ${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [ '3.5', '3.6', '3.7', '3.8' ]
+        os: [ ubuntu-latest, macos-latest]
+      fail-fast: false
+    steps:
+    - uses: actions/checkout@v2
+      name: Checkout the repository
+      with:
+        submodules: recursive
+    - name: Setup python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+        architecture: x64
+    - name: Install Apt dependencies
+      run: sudo apt-get install libboost-dev gfortran
+      if: runner.os == 'Linux'
+    - name: Install Brew dependencies
+      run: brew install boost libomp
+      if: runner.os == 'macOS'
+    - name: Upgrade pip
+      run: python3 -m pip install -U pip setuptools
+    - name: Install Python dependencies
+      run: python3 -m pip install ruamel.yaml scons numpy cython
+    - name: Build Cantera
+      run: python3 `which scons` build -j2
+    - name: Test Cantera
+      run: python3 `which scons` test
+    - name: Run Test Problems
+      run: python3 `which scons` test_problems
+    - name: Run Samples
+      run: python3 `which scons` samples
+
+  # Coverage is its own job because macOS builds of the samples
+  # use Homebrew gfortran which is not compatible for coverage
+  # with XCode clang. Also, turning off optimization really
+  # slows down the tests
```

Comment:
Running the tests with optimization disabled should faster if you use the system BLAS and LAPACK libraries, which is what we do now.